### PR TITLE
[monorepo] update lint stack

### PR DESCRIPTION
**Pull request by Simek**
# Why & How

A combined version of:
* #31571
* #31592

With fixes and corrections needed for all packages in the monorepo.

# Test Plan

All checks are passing when running manually locally. All checks are passing when running SDK checks workflow via `act` locally.

### CODE REVIEW

**Review by douglowder — APPROVED** [review]

File: packages/eslint-config-expo/utils/typescript.js [hunk 1]
```diff
@@ -16,42 +16,11 @@ module.exports = {
...
-              String: {
-                message: 'Use `string` instead.',
-                fixWith: 'string',
-              },
-            },
-            extendDefaults: false,
-          },
-        ],
+        '@typescript-eslint/no-empty-object-type': 'warn',
+        '@typescript-eslint/no-wrapper-object-types': 'warn',
```
Comment (wschurman) [inline packages/eslint-config-expo/utils/typescript.js:23]: 👍

File: packages/eslint-config-universe/README.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # eslint-config-universe
-Shared ESLint configs for Node, Web, and universal Expo projects.
+
+Shared ESLint configs for internal Expo projects.
```
Comment (wschurman) [inline packages/eslint-config-universe/README.md:3]: revert, I think this is used externally as well


```suggestion
Shared ESLint configs for Node, Web, and universal Expo projects.
```

File: packages/eslint-config-universe/package.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "eslint-config-universe",
   "version": "13.0.0",
-  "description": "Shared ESLint configs for universal Expo projects, Node, and web",
+  "description": "Shared ESLint configs for internal Expo projects.",
```
Comment (wschurman) [inline packages/eslint-config-universe/package.json:4]: ditto about this being used externally I believe

File: packages/eslint-config-universe/shared/typescript-analysis.js [hunk 4]
```diff
@@ -11,8 +11,8 @@ module.exports = {
         '@typescript-eslint/no-floating-promises': 'warn',
         '@typescript-eslint/no-for-in-array': 'error',
         '@typescript-eslint/no-misused-promises': ['error', { checksVoidReturn: false }],
-        '@typescript-eslint/no-throw-literal': 'warn',
         '@typescript-eslint/no-unnecessary-type-assertion': 'warn',
+        '@typescript-eslint/only-throw-error': 'warn',
```
Comment (wschurman) [inline packages/eslint-config-universe/shared/typescript-analysis.js:15]: 👍

**Review by wschurman — APPROVED** [review]
A few inline comments, nothing major. Otherwise...

[awesomesauce](https://app.graphite.dev//settings/meme-library?org=expo)

File: packages/eslint-config-universe/README.md (hunk 2 above)
Comment (Simek) [reply packages/eslint-config-universe/README.md:3]: This change has already been accepted once by @ide, and was on the `main`.

AFAIK it's correct statement, and us not being prices in that matter lead to some issues/miscommunication in the past, and lead to creation of [`eslint-config-expo`](https://github.com/expo/expo/tree/main/packages/eslint-config-expo) which is tailored and recommended for external use with Expo.

Refs:
* https://exponent-internal.slack.com/archives/C1QP38NQ5/p1709744141238289?thread_ts=1709741770.192479&cid=C1QP38NQ5
* https://exponent-internal.slack.com/archives/C5ERY0TAR/p1637093462088600

Comment (wschurman) [reply packages/eslint-config-universe/README.md:3]: Ah, got it. Sounds good.

**Review by marklawlor — APPROVED** [review]

**Review by tsapeta — APPROVED** [review]

**Review by gabrieldonadel — APPROVED** [review]

**Pull request merged**